RHIDP-3401 - Release notes for release 1.3 (#559)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/snip-removed-functionality-rhidp-2796.adoc

[id="removed-functionality-rhidp-2796"]
= Breaking changes in Backstage release 1.27

Removed some database helper functions::
The helper functions `createDatabaseClient` and `ensureDatabaseExists` have been removed from the public interface, since they have no usage within the repository and never were suitable for calling from the outside. Switch to using `coreServices.database` or `DatabaseManager` instead.

Deprecated the backend system utilities::
Utilities in the backend system (@backstage/backend-common) are being deprecated with the intent to remove the package entirely in a future release

Removed the requirement to configure a secret for backend authentication::

> The requirement to configure a secret for backend authentication in production has been removed. It is now only needed if you rely on the legacy authentication mechanism (https://backstage.io/docs/auth/service-to-service-auth#external-callers-legacy). If you don’t configure any secrets you will also not be able to generate tokens with the TokenManager service, although use of this service is discouraged as it has been replaced by the AuthService.

Reviewer: ```suggestion
The requirement to configure a secret for backend authentication in production has been removed. It is now only needed if you rely on the legacy authentication mechanism (https://backstage.io/docs/auth/service-to-service-auth#external-callers-legacy). If you do not configure any secrets you will also not be able to generate tokens with the TokenManager service, although use of this service is discouraged as the AuthService has replaced it.

```
